src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -1429,6 +1441,29 @@ class MavlinkStreamVFRHUD : public MavlinkStream
 				msg.climb = -pos.vz;
 			}
 
+			if (status.is_vtol && status.is_vtol_tailsitter && (status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING)) {
+				// This is a tailsitter VTOL flying in fixed wing mode:
+				// report attitude rotated by 90 degrees upward pitch
+				_rotation = get_rot_quaternion(ROTATION_PITCH_90);
+
+			} else {
+				// Normal case
+				_rotation = get_rot_quaternion(ROTATION_NONE);
+			}
+
+			// Attitude
+			const matrix::Quatf att = matrix::Quatf(attitude.q) * _rotation;
+			msg.q1 = att(0);
```

Comment:
I suppose this is a question for https://github.com/mavlink/mavlink/pull/1247, but why isn't the quaternion an array?

Reply:
I used the same format as in ATTITUDE_QUATERNION.
Both should be array in my opinion, or at least 0 indexed...